tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -202,3 +217,224 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
       assert.include hls, anchor12.highlight[anchor12.startPage]
       assert.include hls, anchor2.highlight[anchor2.startPage]
       assert.notInclude hls, anchor3.highlight[anchor3.startPage]
+
+  describe 'two-phased anchoring', ->
+
+    class DummyDocumentAccess
+
+      @applicable: -> true
+
+      isPageMapped: (index) -> index in @_rendered
+
+      # Helper function to trigger a page rendering
+      renderPage: (index) ->
+        return if @isPageMapped index
+
+        @_rendered.push index
+
+        # Publish an event
+        event = document.createEvent "UIEvents"
+        event.initUIEvent "docPageMapped", false, false, window, 0
+        event.pageIndex = index
+        window.dispatchEvent event
+
+      # Helper function to trigger a page unrendering
+      unrenderPage: (index) ->
+        return unless @isPageMapped index
+
+        i = @_rendered.indexOf index
+        @_rendered.splice(i, 1)
+
+        # Publish an event
+        event = document.createEvent "UIEvents"
+        event.initUIEvent "docPageUnmapped", false, false, window, 0
+        event.pageIndex = index
+        window.dispatchEvent event
```

Comment:
On 2015-02-10 13:10, Aron Carroll wrote:
>
> Okay, that's true, but in this case it makes it harder to distinguish
> which properties on the object are mocking the implementation and
> which are part of the test suite.
>
> A mock object will ideally be the simplest thing that acts as a
> stand-in for the real object. Ideally it will just be a plain object
> that has the correct state, or stubs that return the correct values
> for a particular test. The less logic the better.
>

OK, I have separated the helper functions.